projects/RabbitMQ.Client/Impl/RabbitMQActivitySource.cs

```diff
@@ -142,10 +177,22 @@ public static class RabbitMQActivitySource
             return activity;
         }
 
+        internal static void ReportException(this Activity activity, Exception exception)
+        {
+            ActivityTagsCollection exceptionTags = new();
```

Comment:
You should be able to just call `.AddTag` on the current Activity for each tag instead of adding a new event.

Reply:
Opentelemetry spec states that exception should be recorded as event and not tag on the current activity

Reply:
https://opentelemetry.io/docs/specs/otel/trace/exceptions/

Reply:
[System.Diagnostics.DiagnosticSource 9.0.0](https://www.nuget.org/packages/System.Diagnostics.DiagnosticSource/9.0.0) adds a [`AddException`](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.activity.addexception) method to the [`Activity`](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.activity) class.

You can see the implementation [here](https://source.dot.net/#System.Diagnostics.DiagnosticSource/System/Diagnostics/[Activity.cs](https://source.dot.net/#System.Diagnostics.DiagnosticSource/System/Diagnostics/Activity.cs,fac044e024f4be87),fac044e024f4be87).

Although it's safe to add a dependency to version 9.0.0 of this package, if you don't want to risk that, you can add it as an extension method to this project.

Reply:
This recordException is basically a copy paste of the MCR method, I don't know if dependency bump is allowed